src/devices/CharacterLcd/samples/Hd44780.ExtendedSample.cs

```diff
@@ -19,87 +21,73 @@ class ExtendedSample
         const string Fourty = Thirty + "123456789\u000b";
         const string Eighty = Fourty + "123456789\u000c123456789\u000d123456789\u000e123456789\u000f";
 
-        public static void Test()
+        public static void Test(Hd44780 lcd)
         {
             Console.WriteLine("Starting...");
+            lcd.Clear();
+            Console.WriteLine("Initialized");
+            Console.ReadLine();
+            TestPrompt("SetCursor", lcd, SetCursorTest);
+            TestPrompt("Underline", lcd, l => l.UnderlineCursorVisible = true);
+            lcd.UnderlineCursorVisible = false;
+            TestPrompt("Walker", lcd, WalkerTest);
+            CreateTensCharacters(lcd);
+            TestPrompt("CharacterSet", lcd, CharacterSet);
 
-#if USEI2C
-            var i2cDevice = I2cDevice.Create(new I2cConnectionSettings(busId: 1, deviceAddress: 0x21));
-            var controller = new Mcp23008(i2cDevice);
-            var lcd = new Lcd1602(registerSelectPin: 1, enablePin: 2, dataPins: new int[] { 3, 4, 5, 6 }, backlightPin: 7, controller: controller);
-#elif USERGB
-            var i2cLcdDevice = I2cDevice.Create(new I2cConnectionSettings(busId: 1, deviceAddress: 0x3E));
-            var i2cRgbDevice = I2cDevice.Create(new I2cConnectionSettings(busId: 1, deviceAddress: 0x62));
-            var lcd = new LcdRgb1602(i2cLcdDevice, i2cRgbDevice);
-#else
-            Hd44780 lcd = new Hd44780(new Size(20, 4), LcdInterface.CreateGpio(12, 26, new int[] { 16, 17, 18, 19, 20, 21, 22, 23 }, readWritePin: 13));
-#endif
-            using (lcd)
-            {
-                Console.WriteLine("Initialized");
-                Console.ReadLine();
-
-                TestPrompt("SetCursor", lcd, SetCursorTest);
-                TestPrompt("Underline", lcd, l => l.UnderlineCursorVisible = true);
-                lcd.UnderlineCursorVisible = false;
-                TestPrompt("Walker", lcd, WalkerTest);
-                CreateTensCharacters(lcd);
-                TestPrompt("CharacterSet", lcd, CharacterSet);
+            TestPrompt("DisplayEnable", lcd, DisplayAndBackLightOnOff);
 
-                // Shifting
-                TestPrompt("Autoshift", lcd, AutoShift);
-                TestPrompt("DisplayLeft", lcd, l => ShiftDisplayTest(l, a => a.ShiftDisplayLeft()));
-                TestPrompt("DisplayRight", lcd, l => ShiftDisplayTest(l, a => a.ShiftDisplayRight()));
-                TestPrompt("CursorLeft", lcd, l => ShiftCursorTest(l, a => a.ShiftCursorLeft()));
-                TestPrompt("CursorRight", lcd, l => ShiftCursorTest(l, a => a.ShiftCursorRight()));
+            // Shifting
```

Comment:
why are the changes here needed? Would it make sense to instead add separate sample?

Reply:
Part of it is just whitespace, but the main change is that I moved the display initialization code outside, so that the tests can be run against any interface (8 bit, 4 bit, direct or I2C). But I think you are right, I should split this long method up a bit.